exact.go

```diff
@@ -3,104 +3,224 @@
 // roudoff errors.
 package exact
 
+import (
+	"math/big"
+)
+
 type (
 	// Frac is a fraction
 	Frac struct {
-		P int64 // P is the numerator
-		Q int64 // Q is the denominator
+		Sign bool
+		P    *big.Int // P is the numerator
+		Q    *big.Int // Q is the denominator
 	}
 )
 
+var (
+	zero = big.NewInt(0)
+	one  = big.NewInt(1)
+
+	// DefPrecision is the default precision used when
+	// the function does not specify one.
+	DefPrecision Frac
+)
+
+func init() {
+	var prec, _, _ = big.NewFloat(0).Parse("1.0e100", 10)
+	z := big.NewInt(0)
+	prec.Int(z)
+	DefPrecision = NewFrac2(big.NewInt(1), z, false)
+}
+
+// One is the whole.
+func One() Frac { return NewFrac2(one, one, false) }
+
+// Zero is the empty.
+func Zero() Frac { return NewFrac2(zero, one, false) }
+
+// NewFrac creates a new fraction using p and q as numerator and
+// denominator, respectively. Sign tells if it's a negative fraction.
+func NewFrac(p, q uint64, sign bool) Frac {
+	return Frac{
+		Sign: sign,
+		P:    big.NewInt(0).SetUint64(p),
+		Q:    big.NewInt(0).SetUint64(q),
+	}
+}
+
+// NewFrac2 creates a new fraction in the same way as NewFrac
+// but using big.Int as numerator and denominator.
+func NewFrac2(p, q *big.Int, sign bool) Frac {
```

Comment:
This is one of the moments where I think if function overload is that terrible =P

Reply:
Had a very fast idea of having the packages:

* fract
* bigfract

So we can isolate properly the bigint fract stuff from smaller fract stuff. Usually people will choose one in a program and stick with it, no ?

Reply:
No, the idea is always to use big integers internally, because when computing numbers like the infamous pi or computing sqrt of fractions, the numerators/denumerators increase very fast and soon implicitly wrap to `MaxUint16 - valueThatOverflow`. This is something I really disliked in Go, because as it seems a programming error it should not be a good idea to panic instead?

But in the API, the user doesn't need to handle *big.Int for relatively small numbers, but we need another constructor for the cases of very large integers in the initialization. 

Trade offs everywhere: https://golang.org/doc/faq#overloading

Reply:
Hmm I see your point of internally always being big numbers anyway, but you are delivering a two different public API's, not sure if two packages would still be a bad idea (the fract would only call bigfract instead), specially because of the lack of overloading.

Reply:
but there's only one type Frac. I don't think create a new package makes sense only to introduce an alternate way to initialize a Frac.

Reply:
Sorry for some reason I thought there would be a different fract for big int, it is indeed the same type. Perhaps call it NewBigFract ? Fract2 is pretty lame =P (although I understand the pain of naming this)